File: python/discount_group.py
import sdk.python.entities
from sdk.python.entities import Property
from sdk.python.discount import Discount
from sdk.python.product import Product



class DiscountGroup(sdk.python.entities.Entity):

    resource = '/discount_group/'
    json_name = 'discountGroup'

    id = Property(int)
    discount_type = Property(int)
    discounts = Property(Discount)
    product = Property(Product)


class DiscountGroups(sdk.python.entities.Resource):

    entity_class = DiscountGroup
    json_name = 'discountGroups'


discountGroups = DiscountGroups()

File: javascript/discount_group.js
import generateUUID from './uuid.js';
import { addPropertyTo } from './model.js';


export function DiscountGroup() {
    this.resource = '/discountGroups';
    this.json = 'discountGroup';
    this.temporaryId = generateUUID();

    addPropertyTo(this, 'id');
    addPropertyTo(this, 'discountType');
    addPropertyTo(this, 'discounts');
    addPropertyTo(this, 'name');
    addPropertyTo(this, 'product');
}
